rustls/src/time_provider.rs

```diff
@@ -0,0 +1,47 @@
+//! no-std time provider
+
+use alloc::sync::Arc;
+use core::fmt::Debug;
+
+use pki_types::UnixTime;
+
+/// An object that provides the current time
```

Comment:
After the rustdoc headline I think we should mention some of the places where a time provider is used . For certificate validation in particular it's important that the time be accurate or an expired certificate could be accepted. Calling that out explicitly might not hurt.